### PR TITLE
fix: avoid off by one in latest_batch

### PR DESCRIPTION
**Pull request by mpkouznetsov**
## Description

Report latest_batch = 20 after 20 batches.

To make the `while` loop behave identically to a `for batch in range`, increment `batch` at the very end of the loop body.
## Commentary (optional)

Snippet of output from 2_checkpoints.yaml:
```
[2022-05-02T20:42:19.031935Z] 7ebd9bfa || INFO: [1] root: Launching: ['sh', '-c', './2_checkpoints.py']
[2022-05-02T20:42:20.964281Z] 7ebd9bfa || x is now 1
[2022-05-02T20:42:21.064388Z] 7ebd9bfa || x is now 2
[2022-05-02T20:42:21.165811Z] 7ebd9bfa || x is now 3
[2022-05-02T20:42:21.267281Z] 7ebd9bfa || x is now 4
[2022-05-02T20:42:21.367425Z] 7ebd9bfa || x is now 5
[2022-05-02T20:42:21.469374Z] 7ebd9bfa || x is now 6
[2022-05-02T20:42:21.572790Z] 7ebd9bfa || x is now 7
[2022-05-02T20:42:21.675244Z] 7ebd9bfa || x is now 8
[2022-05-02T20:42:21.776249Z] 7ebd9bfa || x is now 9
[2022-05-02T20:42:21.880603Z] 7ebd9bfa || x is now 10
[2022-05-02T20:42:21.881107Z] 7ebd9bfa || INFO:determined.core:report_training_metrics(latest_batch=10, metrics={'x': 10})
[2022-05-02T20:42:21.971129Z] 7ebd9bfa || INFO:determined.core:Reported checkpoint to master 80eb3b3e-c38d-446c-807f-9ffc2b0530ca
```

Snippet of output from 3_hpsearch.yaml:
```
[2022-05-02T20:43:52.448373Z] 9ac955c7 || x is now 8
[2022-05-02T20:43:52.543300Z] 9ac955c7 || x is now 10
[2022-05-02T20:43:52.643485Z] 9ac955c7 || x is now 12
[2022-05-02T20:43:52.743674Z] 9ac955c7 || x is now 14
[2022-05-02T20:43:52.844076Z] 9ac955c7 || x is now 16
[2022-05-02T20:43:52.944260Z] 9ac955c7 || x is now 18
[2022-05-02T20:43:53.045112Z] 9ac955c7 || x is now 20
[2022-05-02T20:43:53.045232Z] 9ac955c7 || INFO:determined.core:report_training_metrics(latest_batch=10, metrics={'x': 20})
[2022-05-02T20:43:53.229947Z] 9ac955c7 || INFO:determined.core:Reported checkpoint to master 064077cd-a4e9-49a3-a22b-d5a43b5aec56
[2022-05-02T20:43:53.330575Z] 9ac955c7 || x is now 22
[2022-05-02T20:43:53.430317Z] 9ac955c7 || x is now 24
[2022-05-02T20:43:53.531869Z] 9ac955c7 || x is now 26
[2022-05-02T20:43:53.632092Z] 9ac955c7 || x is now 28
[2022-05-02T20:43:53.735403Z] 9ac955c7 || x is now 30
[2022-05-02T20:43:53.836382Z] 9ac955c7 || x is now 32
[2022-05-02T20:43:53.936779Z] 9ac955c7 || x is now 34
[2022-05-02T20:43:54.036646Z] 9ac955c7 || x is now 36
[2022-05-02T20:43:54.136748Z] 9ac955c7 || x is now 38
[2022-05-02T20:43:54.236871Z] 9ac955c7 || x is now 40
[2022-05-02T20:43:54.236953Z] 9ac955c7 || INFO:determined.core:report_training_metrics(latest_batch=20, metrics={'x': 40})
[2022-05-02T20:43:54.341972Z] 9ac955c7 || INFO:determined.core:Reported checkpoint to master 5ac273d9-8827-45dc-95dd-fe40d80ef7c9
```

Snippet of output from 4_distributed.yaml:
```
[2022-05-02T20:45:16.444632Z] 954400be || x is now 2912
[2022-05-02T20:45:16.547789Z] 954400be || x is now 2944
[2022-05-02T20:45:16.650758Z] 954400be || x is now 2976
[2022-05-02T20:45:16.752593Z] 954400be || x is now 3008
[2022-05-02T20:45:16.855074Z] 954400be || x is now 3040
[2022-05-02T20:45:16.957434Z] 954400be || x is now 3072
[2022-05-02T20:45:17.057598Z] 954400be || x is now 3104
[2022-05-02T20:45:17.159029Z] 954400be || x is now 3136
[2022-05-02T20:45:17.259584Z] 954400be || x is now 3168
[2022-05-02T20:45:17.362229Z] 954400be || x is now 3200
[2022-05-02T20:45:17.362276Z] 954400be || INFO:determined.core:report_training_metrics(latest_batch=100, metrics={'x': 3200})
[2022-05-02T20:45:17.418133Z] 954400be || INFO:determined.core:Reported checkpoint to master f2efea3c-a037-4708-a9a7-101c7ce556ed
[2022-05-02T20:45:17.419698Z] 954400be || INFO:determined.core:report_validation_metrics(latest_batch=100, metrics={'x': 3200})
[```

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.


<!---
## Title

Example title: "docs: tweak recommended "pip install" usage".

Specifically, this title should contain a type and a description
of the change being made:

User-facing change types:
- docs: docs-only change
- feat: new user-facing feature
- fix: bug fix
- perf: performance improvement

Internal change types:
- build: build system change (anything in a `Makefile`, mostly)
- chore: any internal change not covered by another type
- ci: anything that touches `.circleci`
- refactor: internal refactor
- style: style change
- test: new tests

See https://www.conventionalcommits.org/en/v1.0.0/ for background.

The first line should also:
- be at most 89 characters long
- contain a description that is at most 72 characters long
- not end with sentence-ending punctuation
- start (after the type) with a lowercase imperative ("add", "fix")
-->

### CODE REVIEW

File: examples/tutorials/core_api/2_checkpoints.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def main(core_context, latest_checkpoint, trial_id, increment_by):
         time.sleep(.1)
         print("x is now", x)
         if batch % 10 == 9:
-            core_context.train.report_training_metrics(latest_batch=batch, metrics={"x": x})
+            core_context.train.report_training_metrics(latest_batch=batch+1, metrics={"x": x})
 
             # NEW: write checkpoints at regular intervals to limit lost progress in case of a crash
             # during training.
```
Comment (rb-determined-ai) [inline examples/tutorials/core_api/2_checkpoints.py:61]: you still need a `batch + 1` in save_state in this example (a few lines below this comment)

**Review by rb-determined-ai — APPROVED** [review]
approve, with the one comment

File: examples/tutorials/core_api/2_checkpoints.py (hunk 1 above)
Comment (mpkouznetsov) [reply examples/tutorials/core_api/2_checkpoints.py:61]: yes, just realized that

File: examples/tutorials/core_api/1_metrics.py [hunk 2]
```diff
@@ -22,9 +22,9 @@ def main(core_context, increment_by):
         print("x is now", x)
         # NEW: report training metrics to Determined.
         if batch % 10 == 9:
-            core_context.train.report_training_metrics(latest_batch=batch, metrics={"x": x})
+            core_context.train.report_training_metrics(steps_completed=batch, metrics={"x": x})
```
Comment (rb-determined-ai) [inline examples/tutorials/core_api/1_metrics.py:25]: looks like 1_metrics.py hasn't been updated with `steps_completed = batch + 1` yet

**Review by rb-determined-ai — APPROVED** [review]
My name is rb and I approve this message.

**Pull request merged**